Replace current payout box on bounty page with new design

hshar7 opened the pull request:
- Added new component to hold two currencies.
- Added remaining balance to bounty box.
- Changed design of pill to be next to bounty box.

hshar7 commented:
![ethbox2](https://user-images.githubusercontent.com/17727543/51718622-b2eb3980-200b-11e9-831b-e25c64626bd0.PNG)

c-o-l-o-r commented on src/containers/Bounty/Bounty.module.scss:
```diff
@@ -38,9 +38,10 @@
+  text-align: left;
   height: fit-content;
   white-space: nowrap;
+  border-radius: 5%;
```
Let's remove this and keep the `border-radius` on line 40. In general, it's best to try and use the existing variables.

corwinharrell replied:
@c-o-l-o-r @hshar7 right now I don't think that's something we can help all that much. Shorter titles are going to result in the payout box just taking up the majority of vertical space. In the new fora designs I've adjusted the spacing and we will also be adding additional lines to the title sections (like for indicating which fora community the bounty originates from) so that should help kill some of that extra space. If you want, we could make it so that the numerals and `ETH` elements are the same size (right now the numerals are bigger). That would also decrease the height slightly.

c-o-l-o-r commented on src/containers/Bounty/Bounty.module.scss:
```diff
@@ -66,14 +67,12 @@
   position: absolute;
-  right: 0;
-  top: 0;
+  left: 30%;
```
We should move the stage pill into the `BountyPayout` component -- so you'll also need a new prop to accept the bounty stage.

hshar7 replied:
We'd need to rethink `BountyPayout` component then. If we move the pill to it, it'd have more responsibility than just bounty payouts.

corwinharrell replied:
I agree here, this is one component we'll likely need to refactor a bit, since it encompasses the status now instead of just the value.

c-o-l-o-r commented on src/containers/Bounty/index.js:
```diff
@@ -351,7 +360,7 @@ class BountyComponent extends React.Component {
                         src={`https://ipfs.infura.io/ipfs/${
                           bounty.sourceDirectoryHash
-                        }/${bounty.sourceFileName}`}
+                          }/${bounty.sourceFileName}`}
```
extra space?

hshar7 replied:
my bad!

c-o-l-o-r commented on src/explorer-components/BountyPayoutData/BountyPayoutData.module.scss:
```diff
@@ -0,0 +1,10 @@
+.payoutEthBox {
+    margin-top: 0.5rem;
```
Replace margins with corresponding variables (i.e. `$m-spacing`). Check `styles/variables.scss` for a full list of vars

c-o-l-o-r commented on src/explorer-components/BountyPayoutData/index.js:
```diff
@@ -0,0 +1,90 @@
+import { Currency } from 'explorer-components';
+import { Text } from 'components';
+
+const BountyPayoutData = props => {
```
Let's change this to `BountyPayout`

hshar7 replied:
okay

corwinharrell commented on src/explorer-components/BountyPayoutData/BountyPayoutData.module.scss:
```diff
@@ -0,0 +1,10 @@
+}
+.balanceEthBox {
+    margin-top: 1.5rem;
+    margin-bottom: 0.5rem;
```
If we can, let's try and use our spacing variables whenever possible.

corwinharrell commented on src/explorer-components/BountyPayoutData/BountyPayoutData.module.scss:
```diff
@@ -0,0 +1,10 @@
+    margin-bottom: 0.5rem;
+}
+.currencyTypeHeaderText {
+    font-size: 0.775rem;
```
Same here, let's try and use value's based on our typographic scale. If we can't use a prop, check the variables file for the typographic scale values.

hshar7 commented:
![ethbox3](https://user-images.githubusercontent.com/17727543/51775913-e4700d80-20bc-11e9-985e-de6a20c4aae6.PNG)
![ethboxmobile](https://user-images.githubusercontent.com/17727543/51775914-e4700d80-20bc-11e9-8170-b46b4906eb90.PNG)
Addressed all comments. Attached are desktop and mobile views.

hshar7 commented:
Good to merge?
